src/main/java/io/axoniq/axonserver/connector/event/DcbEventChannel.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2020-2024. AxonIQ
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.axoniq.axonserver.connector.event;
+
+import io.axoniq.axonserver.connector.ResultStream;
+import io.axoniq.axonserver.grpc.event.dcb.AppendEventsResponse;
+import io.axoniq.axonserver.grpc.event.dcb.ConsistencyCondition;
+import io.axoniq.axonserver.grpc.event.dcb.GetHeadRequest;
+import io.axoniq.axonserver.grpc.event.dcb.GetHeadResponse;
+import io.axoniq.axonserver.grpc.event.dcb.GetSequenceAtRequest;
+import io.axoniq.axonserver.grpc.event.dcb.GetSequenceAtResponse;
+import io.axoniq.axonserver.grpc.event.dcb.GetTagsRequest;
+import io.axoniq.axonserver.grpc.event.dcb.GetTagsResponse;
+import io.axoniq.axonserver.grpc.event.dcb.GetTailRequest;
+import io.axoniq.axonserver.grpc.event.dcb.GetTailResponse;
+import io.axoniq.axonserver.grpc.event.dcb.SourceEventsRequest;
+import io.axoniq.axonserver.grpc.event.dcb.SourceEventsResponse;
+import io.axoniq.axonserver.grpc.event.dcb.StreamEventsRequest;
+import io.axoniq.axonserver.grpc.event.dcb.StreamEventsResponse;
+import io.axoniq.axonserver.grpc.event.dcb.TaggedEvent;
+
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Communication channel for Event related interaction with Axon Server based on Dynamic Consistency Boundaries
+ * concept.
+ *
+ * @author Milan Savic
+ * @since 2024.1.0
+ */
+public interface DcbEventChannel {
+
+    /**
+     * Starts a new transaction to append events.
+     *
+     * @return the transaction reference onto which to register events to append along with the condition
+     */
+    AppendEventsTransaction startTransaction();
```

Comment:
Could we change this to expect the `ConsistencyCondition` as a parameter instead? Or, add another method that take the condition, adjusting this version to clarify it is a "condition-less" append operation (which is perfectly valid too, of course).

Reply:
I prefer an overload of the method, since the `condition` is not required - you can append events unconditionally to the Event Store. However, that would require a different object to be returned, since it wouldn't make sense to change the `condition` once passed to the method. wdyt?

Reply:
Hmm that could work! What we've done in AF5 recently, was add builders that switch to different states. The test fixtures in AF are also an example of this. So, you could have an append-transaction flow, that for **this** operation returns the start of the builder and for the one taking in a `condition` starts at the "append events"-phase instead.